### PR TITLE
Add vary header length patch

### PR DESCRIPTION
**Pull request by es**
Ports https://github.com/Shopify/nginx-shopify-build/pull/42 to dev Nginx.

r: @pushrax @csfrancis

### CODE REVIEW

File: nginx-shopify.rb [hunk 1]
```diff
@@ -27,6 +27,11 @@ def self.nginx_modules
     sha256 "73f29eb281e7da95f4db3b33e44d318e685fef7328e79c6c01a7393f88cef788"
   end
 
+  patch do
+    url "https://gist.githubusercontent.com/es/ff969cad8f7461cffe285954025d0901/raw/082ae31d0e0b1b6487eb03e52b38dd215ef7a0fc/vary_header_length.patch"
+    sha256 "b5125816f824c2a8048765bff6ae71bc757424ee9387d3de759100278eeffdef"
+  end
+
```
Comment (pushrax) [inline nginx-shopify.rb]: LINUX WHEN

File: vary_header_length.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--- a/src/http/ngx_http_cache.h 2015-06-05 15:10:36.000000000 +0100
++++ b/src/http/ngx_http_cache.h 2015-06-05 15:16:21.000000000 +0100
+@@ -25,7 +25,7 @@
+
+ #define NGX_HTTP_CACHE_KEY_LEN       16
+ #define NGX_HTTP_CACHE_ETAG_LEN      42
+-#define NGX_HTTP_CACHE_VARY_LEN      42
++#define NGX_HTTP_CACHE_VARY_LEN      256
+
+ #define NGX_HTTP_CACHE_VERSION       3
```
Comment (es) [inline vary_header_length.patch]: Shoot sorry, accidentally added that to the commit.

File: nginx-shopify.rb (hunk 1 above)
Comment (es) [inline nginx-shopify.rb]: I was thinking we should move to Windows :trollface:

**Comment by es** [comment]
Ready for another round of 👀

File: nginx-shopify.rb (hunk 1 above)
Comment (csfrancis) [inline nginx-shopify.rb]: #patcheswelcome

**Pull request merged**